kernel/src/features/mod.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+
+use crate::Error;
+
+pub use column_mapping::ColumnMappingMode;
+pub(crate) use column_mapping::COLUMN_MAPPING_MODE_KEY;
+
+mod column_mapping;
+
+/// Features table readers can support as well as let users know
+/// what is supported
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub enum ReaderFeatures {
+    /// Mapping of one column to another
+    ColumnMapping,
+    /// Deletion vectors for merge, update, delete
+    DeletionVectors,
+    /// timestamps without timezone support
+    #[serde(rename = "timestampNtz")]
+    TimestampWithoutTimezone,
+    /// version 2 of checkpointing
+    V2Checkpoint,
+}
+
+impl TryFrom<String> for ReaderFeatures {
+    type Error = Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        value.as_str().try_into()
+    }
+}
+
+impl TryFrom<&str> for ReaderFeatures {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let val = match value {
+            "columnMapping" => ReaderFeatures::ColumnMapping,
+            "deletionVectors" => ReaderFeatures::DeletionVectors,
+            "timestampNtz" => ReaderFeatures::TimestampWithoutTimezone,
+            "v2Checkpoint" => ReaderFeatures::V2Checkpoint,
```

Comment:
This replication of magic (string) constants makes me nervous -- no compile-time protection against typos or omissions. Do we have complete round trip unit tests which could somehow catch both present and future typos/incompleteness in this list?

Reply:
Actually, we might want to consider using the strum crate, which has derive macros for both [AsRef](https://docs.rs/strum/latest/strum/derive.AsRefStr.html) and [FromStr+TryFrom](https://docs.rs/strum/latest/strum/derive.EnumString.html).